packages/eslint-plugin/src/rules/ban-tslint-comment.ts

```diff
@@ -0,0 +1,60 @@
+import { AST_TOKEN_TYPES } from '@typescript-eslint/experimental-utils';
+import * as util from '../util';
+
+// tslint regex
+// https://github.com/palantir/tslint/blob/95d9d958833fd9dc0002d18cbe34db20d0fbf437/src/enableDisableRules.ts#L32
+const ENABLE_DISABLE_REGEX = /^\s*tslint:(enable|disable)(?:-(line|next-line))?(:|\s|$)/;
+
+const toText = (
+  text: string,
+  type: AST_TOKEN_TYPES.Line | AST_TOKEN_TYPES.Block,
+): string =>
+  type === AST_TOKEN_TYPES.Line
+    ? ['//', text.trim()].join(' ')
+    : ['/*', text.trim(), '*/'].join(' ');
+
+export default util.createRule({
+  name: 'ban-tslint-comment',
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'Bans `// tslint:<rule-flag>` comments from being used',
+      category: 'Stylistic Issues',
+      recommended: false,
+    },
+    messages: {
+      commentDetected: 'tslint comment detected: "{{ text }}"',
+    },
+    schema: [],
+    fixable: 'code',
+  },
+  defaultOptions: [],
+  create: context => {
+    const sourceCode = context.getSourceCode();
+    return {
+      Program(): void {
+        const comments = sourceCode.getAllComments();
+        comments.forEach(c => {
+          if (ENABLE_DISABLE_REGEX.test(c.value)) {
+            context.report({
+              data: { text: toText(c.value, c.type) },
+              node: c,
+              messageId: 'commentDetected',
+              fix(fixer) {
+                const rangeStart = sourceCode.getIndexFromLoc({
+                  column: c.loc.start.column > 0 ? c.loc.start.column - 1 : 0,
+                  line: c.loc.start.line,
+                });
+                const rangeEnd = sourceCode.getIndexFromLoc({
+                  column: c.loc.end.column,
+                  line: c.loc.end.line,
+                });
+                return fixer.removeRange([rangeStart, rangeEnd + 1]);
```

Comment:
why not just delete the entire comment?
`fixer.remove(c)`

Reply:
:facepalm: I... don’t know. Fixing it now.

Reply:
Actually. I do remember - I think remove() only removed the comment and left behind and empty line (and white space). This contrived removeRange fixed it (anecdotally). I’m still happy to flip it back though. What do you think?

Reply:
@bradzacher if I  use `remove()`, I end up getting these test failures. I can definitely switch the expected output back (and maybe that makes more sense, since the extra whitespace will probably get handled by other rules?). Let me know if I should leave this as-is, or switch to `remove()` and update the test cases.

```
 FAIL  tests/rules/ban-tslint-comment.test.ts
  ban-tslint-comment
    valid
      ✓ let a: readonly any[] = []; (180ms)
      ✓ let a = new Array(); (3ms)
      ✓ // some other comment (3ms)
      ✓ // TODO: this is a comment that mentions tslint (1ms)
      ✓ /* another comment that mentions tslint */ (1ms)
    invalid
      ✓ /* tslint:disable */ (6ms)
      ✓ /* tslint:enable */ (1ms)
      ✓ /* tslint:disable:rule1 rule2 rule3... */ (1ms)
      ✓ /* tslint:enable:rule1 rule2 rule3... */ (2ms)
      ✓ // tslint:disable-next-line (2ms)
      ✕ someCode(); // tslint:disable-line (8ms)
      ✓ // tslint:disable-next-line:rule1 rule2 rule3... (2ms)
      ✕ const woah = doSomeStuff();
// tslint:disable-line
console.log(woah);
 (4ms)

  ● ban-tslint-comment › invalid › someCode(); // tslint:disable-line

    assert.strictEqual(received, expected)

    Expected value to strictly be equal to:
      "someCode();"
    Received:
      "someCode(); "
    
    Message:
      Output is incorrect.

    Difference:

    - Expected
    + Received

    - someCode();
    + someCode(); 

      at testInvalidTemplate (../../node_modules/eslint/lib/rule-tester/rule-tester.js:845:28)
      at Object.<anonymous> (../../node_modules/eslint/lib/rule-tester/rule-tester.js:874:25)

  ● ban-tslint-comment › invalid › const woah = doSomeStuff();
// tslint:disable-line
console.log(woah);


    assert.strictEqual(received, expected)

    Expected value to strictly be equal to:
      "const woah = doSomeStuff();
    console.log(woah);
    "
    Received:
      "const woah = doSomeStuff();·
    console.log(woah);
    "
    
    Message:
      Output is incorrect.

    Difference:

    - Expected
    + Received

      const woah = doSomeStuff();
    +
      console.log(woah);

      at testInvalidTemplate (../../node_modules/eslint/lib/rule-tester/rule-tester.js:845:28)
      at Object.<anonymous> (../../node_modules/eslint/lib/rule-tester/rule-tester.js:874:25)


=============================== Coverage summary ===============================
Statements   : 1.51% ( 77/5096 )
Branches     : 0.17% ( 8/4576 )
Functions    : 0.78% ( 8/1028 )
Lines        : 1.46% ( 74/5059 )
================================================================================
Test Suites: 1 failed, 1 total
Tests:       2 failed, 11 passed, 13 total
Snapshots:   0 total
Time:        6.278s
Ran all test suites matching /ban-tslint/i.
```